src/app/shared/entry.ts

```diff
@@ -202,10 +203,26 @@ export abstract class Entry implements OnInit, OnDestroy {
       if (foundVersion) {
         return foundVersion;
       }
+    } else {
```

Comment:
Since you are potentially returning null, the function signature should indicate that it might return null.

Previously, I wouldn't even return null and let it return undefined, but I guess with the null checker we want to to turn on, it probably is better to be explicit.